opa/policies/conflicting/autoscaler_service_modification.rego

```diff
@@ -0,0 +1,127 @@
+# schemas:
+#   - input: schema.terraform.plan
+# related_resources:
+#   - https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/service_integration
+#   - https://aiven.io/docs/platform/howto/add-storage-space
+deny contains msg if {
+	# Get services that are losing autoscaler
+	service_name := services_losing_autoscaler[_]
+
+	# Check if the same service is also being modified
+	service_name in services_being_modified
+
+	msg := sprintf(
+		concat("", [
+			"POLICY VIOLATION: Autoscaler integration removal and service modification conflict detected. ",
+			"Service '%s' has an autoscaler integration being removed while the service itself is being modified in the same apply. ",
+			"This can cause inconsistent Terraform plans due to disk space management conflicts. ",
+			"Please separate these operations: ",
+			"First, remove the autoscaler integration alone. ",
+		]),
+		[service_name],
+	)
+}
+
+# Deny rule for configuration-level detection
+deny contains msg if {
+	# Find autoscaler integrations in configuration that are not in planned values (being removed)
+	integration_config := input.configuration.root_module.resources[_]
+	integration_config.type == "aiven_service_integration"
+
+	# get the integration configuration expressions
+	config_integration_type := integration_config.expressions.integration_type.constant_value
+	config_integration_type == "autoscaler"
+
+	# get the source service name from configuration
+	source_service_ref := integration_config.expressions.source_service_name.references[0]
+
+	# Find if this autoscaler integration is being removed (not in planned values)
+	integration_being_removed := is_integration_being_removed(integration_config.address)
+	integration_being_removed == true
+
+	# check if any service resource references the same service and is being modified
+	service_config := input.configuration.root_module.resources[_]
+	startswith(service_config.type, "aiven_")
+	not service_config.type == "aiven_service_integration"
+	not service_config.type == "aiven_service_integration_endpoint"
+
+	# check if this service resource is the one referenced by the autoscaler
+	service_resource_name := sprintf("%s.%s", [service_config.type, service_config.name])
+	service_resource_name == source_service_ref
+
+	# check if this service is being modified
+	service_being_modified := is_service_being_modified(service_config.address)
+	service_being_modified == true
+
+	msg := sprintf(
+		concat("", [
+			"POLICY VIOLATION: Autoscaler integration removal and service modification conflict detected. ",
+			"Service resource '%s' is being modified while its autoscaler integration '%s' is being removed in the same apply. ",
+			"This can cause inconsistent Terraform plans due to disk space management conflicts. ",
+			"Please separate these operations: ",
+			"1) First, remove the autoscaler integration alone, or ",
+			"2) Add explicit 'additional_disk_space' configuration before removing the autoscaler.",
```

Comment:
I don't think it is possible to add `additional_disk_space` while the autoscaler exist. There is a check for that https://github.com/aiven/terraform-provider-aiven/blob/c48ba11ef30741ee027c7112c0bbb3c0c59562b2/internal/schemautil/custom_diff.go#L304